imagebuildah/stage_executor.go

```diff
@@ -1164,6 +1164,30 @@ func (s *StageExecutor) Execute(ctx context.Context, base string) (imgID string,
 		// determining if a cached layer with the same build args already exists
```

Comment:
The commend above about ARG is also suggestive that there might be some more logic to this.